web/static/js/services/retro_channel.js

```diff
@@ -47,8 +47,9 @@ const applyListenerCallbacks = (retroChannel, store, actions) => {
     actions.deleteIdea(deletedIdea.id)
   })
 
-  retroChannel.on("vote_submitted", votedOnIdea => {
-    actions.updateIdea(votedOnIdea.id, { vote_count: votedOnIdea.vote_count })
+  retroChannel.on("vote_submitted", data => {
+    actions.updateIdea(data.idea.id, { vote_count: data.idea.vote_count })
```

Comment:
Hmmm... not suggesting we change it now, but this dual action dispatch feels dirty, in the same way that making two database updates feels dirty to represent one action, in that our voting state is spread across two tables in the db and now two objects on the store. If this continues to propagate, I have a feeling we're going to regret not properly normalizing and introducing a `votes` table might come back to bite us.

Reply:
@qlaire @ams340 while you're on the bench, can you keep your noses primed for this dual state/action spell propagating throughout the app. @kpthomas pinging you as well, as i'm guessing you thought of this when you did the initial persistence work with Claire a couple weeks ago.

Reply:
Yes will do!